<?xml version="1.0" encoding="utf-8"?>
<launch>

	<!-- Node configuration -->
	<arg name="node_name" default="yolov8" />

	<!-- Input RGB topic -->
	<!--<arg name="image_topic" default="/xtionPeople/rgb/image_raw" />-->
	<arg name="image_topic" default="/xtion/rgb/image_raw" />
	<arg name="image_transport" default="compressed" />

	<!-- Path configuration -->
	<arg name="data_dir" default="$(find upo_nhoa_perception)/data" />
	<arg name="cuda_dir" default="/usr/local/cuda/lib64" />

	<!-- Parameters -->
	<arg name="model_file" default="yolov8m.onnx" />
	<arg name="obj_names_file" default="coco.bin" />
	<arg name="obj_score_thresh" default="0.5" />
	<arg name="nms_thresh" default="0.5" />
	<arg name="filter_on_object" default="" />

	<!-- YOLOv8 object detection node -->
	<node pkg="upo_nhoa_perception" name="$(arg node_name)" type="yolov8_host" output="screen" required="true">
		<env name="LD_LIBRARY_PATH" value="$(env LD_LIBRARY_PATH):$(arg cuda_dir)"/>
		<param name="image_topic" value="$(arg image_topic)" />
		<param name="image_transport" value="$(arg image_transport)" />
		<param name="model_file" value="$(arg data_dir)/$(arg model_file)" />
		<param name="obj_names_file" value="$(arg data_dir)/$(arg obj_names_file)" />
		<param name="obj_score_thresh" value="$(arg obj_score_thresh)" />
		<param name="nms_thresh" value="$(arg nms_thresh)" />
		<param name="filter_on_object" value="$(arg filter_on_object)" />
	</node>

</launch>
